distributed_shampoo/utils/shampoo_preconditioner_list.py

```diff
@@ -1378,21 +1404,15 @@ def _create_kronecker_factors_list(
         )
 
     def _get_inverse_roots_from_override(
-        self, preconditioned_dims_selector_list: tuple[tuple[bool, ...], ...]
-    ) -> tuple[tuple[float, ...], ...]:
+        self, preconditioned_dims_selector: tuple[bool, ...]
+    ) -> tuple[float, ...]:
         # NOTE: In eigenvalue-corrected Shampoo, there is only a single inverse root that is applied to the corrected eigenvalues.
-        return tuple(
-            (
-                # Compute the inverse root, 1 / eigenvalue_inverse_exponent{_override}, for each block.
-                1
-                / getattr(
-                    self._preconditioner_config,
-                    "inverse_exponent_override",
-                    {},
-                ).get(len(preconditioned_dims_selector), 1 / 2),
-            )
-            # Traverse through each block's preconditioned_dims_selector.
-            for preconditioned_dims_selector in preconditioned_dims_selector_list
+        return (
+            # Compute the inverse root, 1 / eigenvalue_inverse_exponent{_override}.
+            1
+            / attrgetter("inverse_exponent_override")(self._preconditioner_config).get(
+                len(preconditioned_dims_selector), 1 / 2
```

Comment:
Could be factored out to overwrite the `_get_inverse_exponent_override_based` method if it was moved to the base class.